### PR TITLE
Feat/setup-fork: Add config file, new parameters, automate kubectl configuration

### PR DESCRIPTION
**Pull request by gilmoregrills**
### Big PR:
- Updated all the local import paths and stuff like that to newscorp-ghfb from jetstack  
- Added the `--kubeconfig` and `--username` parameters and methods to use them
- Automated the browser-opening and kubectl-configuring steps using the above parameters
- Added the ability to pass in a config file instead of adding the id/secret/uri parameters every time

### CODE REVIEW

**Comment by gilmoregrills** [comment]
this project was set up to use [goreleaser](https://goreleaser.com/) to handle releases, so once this is merged we can try using that to build all the binaries and push them to the releases tab here

**Comment by gilmoregrills** [comment]
@marcincuber tested it a bunch and it actually works now! Ready for review 🎉

File: Makefile [hunk 1]
```diff
@@ -59,7 +59,7 @@ docker_push: docker_build
 go_verify: go_fmt go_vet go_test
 
 go_build:
-	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -tags netgo -ldflags '-w -X main.version=$(CI_COMMIT_TAG) -X main.commit=$(CI_COMMIT_SHA) -X main.date=$(shell date -u +%Y-%m-%dT%H:%M:%SZ)' -o okta-kubectl-auth_linux_amd64
```
Comment (marcincuber) [inline Makefile]: Any chances to build both binaries or have steps for both i.e go_build_mac and go_build_linux?

File: README.md [hunk 2]
```diff
@@ -8,6 +8,8 @@ For instructions on how to set up an Okta application that uses the OIDC flow to
 
 ### okta-kubectl-auth
 
+####Running the tool:
```
Comment (marcincuber) [inline README.md]: fix formatting

File: README.md [hunk 3]
```diff
@@ -16,14 +18,39 @@ Once you have compiled and installed `okta-kubectl-auth` and created your Okta a
 We can now run `okta-kubectl-auth` as follows:
 
 ```
-okta-kubectl-auth --client-id=<client-id> --client-secret=<client-secret> --base-domain=<issuer-uri>
+okta-kubectl-auth --client-id=<client-id> --client-secret=<client-secret> --base-domain=<issuer-uri> --kubeconfig <kubeconfig_path> --username <username>
```
Comment (marcincuber) [inline README.md]: Can we clarify bit better what username is as generally it is misleading? Or perhaps some valid examples

File: README.md [hunk 4]
```diff
@@ -16,14 +18,39 @@ Once you have compiled and installed `okta-kubectl-auth` and created your Okta a
...
+        "secret": "<secret>",
+        "uri": "<base-domain>"
+    }
+}
 ```
 
-Follow the instructions printed by `okta-kubectl-auth` to complete the setup process.
+This config file can hold as many config objects as necessary, as long as they all follow this structure.
+
+
```
Comment (marcincuber) [inline README.md]: remove empty lines

File: cmd/root.go [hunk 5]
```diff
@@ -15,6 +18,9 @@ type Flags struct {
 	ClientSecret string
 	BaseDomain   string
 	BindAddr     string
+	InputConfig  string
+    Username     string
```
Comment (marcincuber) [inline cmd/root.go]: indent

File: cmd/root.go [hunk 6]
```diff
@@ -44,22 +50,43 @@ func Execute() {
 
 func newOkta(flags *Flags) *okta.Okta {
 	o := okta.New(nil, flags.Debug)
-	o.BaseDomain = flags.BaseDomain
+	if flags.ClientID == "" {
+	    o.ClientID, o.ClientSecret, o.BaseDomain = ParseConfig(flags.InputConfig, flags.Username)
+	} else {
+	    o.ClientID = flags.ClientID
+        o.ClientSecret = flags.ClientSecret
```
Comment (marcincuber) [inline cmd/root.go:78]: indent

File: pkg/okta/okta.go [hunk 7]
```diff
@@ -202,15 +209,47 @@ func (o *Okta) Authorize(authCodeURLCh chan string) error {
...
 
 func (o *Okta) printKubectlConfiguration(refreshToken string) {
 
-	fmt.Printf("\nRun the following command (replacing <username> with a user in your kubeconfig) to configure kubectl for OIDC authentication:\n\nkubectl config set-credentials \\\n  --auth-provider=oidc \\\n  --auth-provider-arg=idp-issuer-url=%s \\\n  --auth-provider-arg=client-id=%s \\\n  --auth-provider-arg=client-secret=%s \\\n  --auth-provider-arg=refresh-token=%s \\\n  <username>\n", o.BaseDomain, o.ClientID, o.ClientSecret, refreshToken)
+	fmt.Printf("\nRun the following command (replacing <username> with a user in your kubeconfig if necessary) to configure kubectl for OIDC authentication:\n\nkubectl config set-credentials \\\n  --auth-provider=oidc \\\n  --auth-provider-arg=idp-issuer-url=%s \\\n  --auth-provider-arg=client-id=%s \\\n  --auth-provider-arg=client-secret=%s \\\n  --auth-provider-arg=refresh-token=%s \\\n  <username> --kubeconfig <kubeconfig_path>\n", o.BaseDomain, o.ClientID, o.ClientSecret, refreshToken)
+
+	return
+}
+
+func (o *Okta) executeKubectlConfiguration(refreshToken string) {
```
Comment (marcincuber) [inline pkg/okta/okta.go:229]: Fix formatting in this function

File: cmd/root.go [hunk 8]
```diff
@@ -44,22 +50,43 @@ func Execute() {
...
+	RootCmd.Flags().StringVar(&flags.ClientSecret, "client-secret", "", "OAuth2 client secret of this application.")
 
-	RootCmd.PersistentFlags().StringVar(&flags.BaseDomain, "base-domain", "", "URL of the OpenID Connect issuer. (required)")
-	RootCmd.MarkPersistentFlagRequired("base-domain")
+	RootCmd.PersistentFlags().StringVar(&flags.BaseDomain, "base-domain", "", "URL of the OpenID Connect issuer.")
 	RootCmd.PersistentFlags().StringVar(&flags.BindAddr, "bind-addr", "127.0.0.1:8888", "HTTP address to listen at.")
 	RootCmd.PersistentFlags().BoolVar(&flags.Debug, "debug", false, "Raise log level to debug.")
+
+	RootCmd.PersistentFlags().StringVar(&flags.KubeConfig, "kubeconfig", "", "Path to the kubeconfig you want to update.")
+	RootCmd.PersistentFlags().StringVar(&flags.InputConfig, "config", "cluster_config.json", "Path to a json file containing the required keys/tokens. (see README)")
```
Comment (marcincuber) [inline cmd/root.go]: You shouldn't have a default value set to cluster_config.json. I think this is breaking cli in terms of error handling. 
Also take into account that currently only required flag(s) are "base-domain", "client-id", "client-secret". 

I would expect that you also need to better handle parsing of the file in newOkta() function.

**Review by marcincuber — CHANGES_REQUESTED** [review]
See my comments. In general looks good. Error handling doesn't work as expected so take a look into it. One example is execution of okta-kubectl-auth on its own; returns an error because file cluster_config.json doesn't exist.

Also, in terms of docs please add commands how to pull and build the cli locally.

**Review by marcincuber — APPROVED** [review]

**Pull request merged**